sdks/python/apache_beam/runners/interactive/pipeline_fragment.py

```diff
@@ -233,3 +234,9 @@ def visit_transform(self, transform_node):
     v = PruneVisitor()
     runner_pipeline.visit(v)
     return runner_pipeline
+
+
+def should_skip_pruning(transform: AppliedPTransform):
+  return (
+      isinstance(transform.transform, TestStream) or
+      '_DataFrame_' in transform.full_label)
```

Comment:
can we match on is instance `DataframeTransform` instead of string matching?

Reply:
DataframeTransform expands into a bunch of normal ParDos, so DataframeTransform itself doesn't work.

Reply:
Ack, if there's a minimally invasive change we can make in the DataFrame API that would allow this filter be more selective, I'd be in favor of making that here. But that doesn't need to be a blocker.

Reply:
Also - it might be good to make sure this works for operations on DeferredSeries objects as well (i.e. do we need to match on "_Series"?). This could be challenging though. I don't think we've encountered an example of a series getting pruned.

Reply:
I would recommend adding unittests with ib.collect for each Series operation. In that case, we don't have to skip pruning prematurely.

The same also applies to the DeferredDataFrame operations. If applicable, in the future add some unittest with ib.collect (following the InteractiveDataFrameTest in this PR).